.github/workflows/testing.yml

```diff
@@ -31,51 +31,58 @@ jobs:
           - 27017:27017
 
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0} # enables conda/mamba env activation by reading bash profile
     strategy:
       matrix:
         python-version: ["3.9", "3.10", "3.11"]
 
     steps:
-      - uses: actions/checkout@v4
+      - name: Check out repo
+        uses: actions/checkout@v4
 
-      - uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
+      - name: Set up micromamba
```

Comment:
@utf i think it could help to have the `micromamba`-dependent MD CI be its own job or even its own `test-md.yml` workflow. that would enable only running the MD tests when any MD source files change and would also uncouple the remaining CI from any install/env issues `micromamba` might encounter

Reply:
I'd be happy with that solution!

Reply:
Using micromamba saves us from manually building enumlib so I favor keeping it in the main CI.